src/main/java/net/sf/mzmine/modules/peaklistmethods/identification/ms2search/Ms2SearchTask.java

```diff
@@ -0,0 +1,280 @@
+       return this.score;
+   }
+   
+   public int getNumIonsMatched()
+   {
+       return this.numIonsMatched;
+   }
+   
+   public String getSearchType()
+   {
+       return this.searchType;
+   }
+   
+               
+   
+}
+
+public class Ms2SearchTask extends AbstractTask {
+
+    private Logger logger = Logger.getLogger(this.getClass().getName());
+
+    private int finishedRows, totalRows;
+    private PeakList peakList1;
+    private PeakList peakList2;
+
+    private RTTolerance rtTolerance;
+    private MZTolerance mzTolerance;
+    private double maxComplexHeight;
+    private IonizationType ionType;
+    private ParameterSet parameters;
+    private double scoreThreshold;
+    private double intensityThreshold;
+    private int minimumIonsMatched;
+
+    /**
+     * @param parameters
+     * @param peakList
+     */
+    public Ms2SearchTask(ParameterSet parameters, PeakList peakList1,
+            PeakList peakList2) {
+
+        this.peakList1 = peakList1;
+        this.peakList2 = peakList2;
+        this.parameters = parameters;
+
+        mzTolerance = parameters.getParameter(Ms2SearchParameters.mzTolerance)
+                .getValue();
+        
+        scoreThreshold = parameters.getParameter(Ms2SearchParameters.scoreThreshold)
+                .getValue();
+        
+        intensityThreshold = parameters.getParameter(Ms2SearchParameters.intensityThreshold)
+                .getValue();
+        
+        minimumIonsMatched = parameters.getParameter(Ms2SearchParameters.minimumIonsMatched)
+                .getValue();
+
+    }
+
+    /**
+     * @see net.sf.mzmine.taskcontrol.Task#getFinishedPercentage()
+     */
+    public double getFinishedPercentage() {
+        if (totalRows == 0)
+            return 0;
+        return ((double) finishedRows) / totalRows;
+    }
+
+    /**
+     * @see net.sf.mzmine.taskcontrol.Task#getTaskDescription()
+     */
+    public String getTaskDescription() {
+        return "MS2 similarity comparison between " + peakList1 + "and"
+                + peakList2;
+    }
+
+    /**
+     * @see java.lang.Runnable#run()
+     */
+    public void run() {
+
+        setStatus(TaskStatus.PROCESSING);
+
+        logger.info("Starting MS2 similarity search between " + peakList1 + "and"
+                + peakList2 + "with mz tolerance:"+mzTolerance.getPpmTolerance());
+
+        Ms2SearchResult searchResult;
+        double score;
+        int numIonsMatched;
+        PeakListRow rows1[] = peakList1.getRows();
+        PeakListRow rows2[] = peakList2.getRows();
+        
+        int rows1Length = rows1.length;
+        int rows2Length = rows2.length;
+        
+        totalRows = rows1Length;
+        
+        for (int i = 0; i < rows1Length; i++) {
+            for (int j = 0; j < rows2Length; j++) {
+                Feature featureA = rows1[i].getBestPeak();
+                Feature featureB = rows2[j].getBestPeak();
+                
+                int featureAID = featureA.hashCode();
```

Comment:
Was attempting to get a unique identifier per feature, as in the future it may be better to store exactly what feature was searched against & what the results were.  This gets more complicated with the multiple features per single row in the feature list.